Foundation/XMLNode.swift

```diff
@@ -790,6 +793,11 @@ open class XMLNode: NSObject, NSCopying {
         withUnretainedReference {
             _CFXMLNodeSetPrivateData(_xmlNode, $0)
         }
+        if let documentPtr = _CFXMLNodeGetDocument(_xmlNode) {
+            if documentPtr != ptr {
+                _xmlDocument = XMLDocument._objectNodeForNode(documentPtr)
```

Comment:
Won't this create a cyclic strong reference? The document has a reference to its children, and children have a reference to the document. The objects will never be deallocated unless you detach all the children to break the cycle.